libraries/RC_Channel/RC_Channels.cpp

```diff
@@ -217,6 +217,22 @@ void RC_Channels::read_mode_switch()
     c->read_mode_switch();
 }
 
+// check if flight mode channel is assigned RC option
+// return true if assigned
+bool RC_Channels::check_flight_mode_channel_conflict(uint8_t &num)
+{
+    num = flight_mode_channel_number();
+    const RC_Channel *c = channel(num-1);
+    if (c == nullptr) {
+        return false;
+    }
+    const uint16_t option = c->option.get();
+    if (option != 0) {
```

Comment:
This should be using the enumeration value, even if a little casting is required (cast `option`, not the enumeration value)

Reply:
I'd also invert this to be `==` to preserve the `return false;`-early pattern.

Reply:
thank you.
I fixed it.